tests/suite/longevity_test.go

```diff
@@ -0,0 +1,137 @@
+			"longevity/cafe-secret.yaml",
+			"longevity/gateway.yaml",
+			"longevity/cafe-routes.yaml",
+			"longevity/cronjob.yaml",
+		}
+		promFile = []string{
+			"longevity/prom.yaml",
+		}
+
+		ns = &core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "longevity",
+			},
+		}
+
+		labelFilter = GinkgoLabelFilter()
+	)
+
+	BeforeEach(func() {
+		if !strings.Contains(labelFilter, "longevity") {
+			Skip("skipping longevity test unless 'longevity' label is explicitly defined when running")
+		}
+	})
+
+	It("sets up the longevity test", Label("longevity-setup"), func() {
+		if !strings.Contains(labelFilter, "longevity-setup") {
+			Skip("'longevity-setup' label not specified; skipping...")
+		}
+
+		Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(promFile, ngfNamespace)).To(Succeed())
+		Expect(resourceManager.WaitForAppsToBeReady(ns.Name)).To(Succeed())
+	})
+
+	It("collects results", Label("longevity-teardown"), func() {
+		if !strings.Contains(labelFilter, "longevity-teardown") {
+			Skip("'longevity-teardown' label not specified; skipping...")
+		}
+
+		resultsDir, err := framework.CreateResultsDir("longevity", version)
+		Expect(err).ToNot(HaveOccurred())
+
+		filename := filepath.Join(resultsDir, fmt.Sprintf("%s.md", version))
+		resultsFile, err := framework.CreateResultsFile(filename)
+		Expect(err).ToNot(HaveOccurred())
+		defer resultsFile.Close()
+
+		Expect(framework.WriteSystemInfoToFile(resultsFile, clusterInfo, *plusEnabled)).To(Succeed())
+
+		// gather wrk output
+		homeDir, err := os.UserHomeDir()
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(framework.WriteContent(resultsFile, "\n## Traffic\n"))
+		writeTrafficResults(resultsFile, homeDir, "coffee.txt", "HTTP")
+		writeTrafficResults(resultsFile, homeDir, "tea.txt", "HTTPS")
+
+		// gather any error logs
+		names, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(framework.WriteContent(resultsFile, "\n## Error Logs\n"))
+		writeErrorLogs(resultsFile, names[0], "nginx-gateway")
+		writeErrorLogs(resultsFile, names[0], "nginx")
+
+		Expect(resourceManager.DeleteFromFiles(files, ns.Name)).To(Succeed())
+		Expect(resourceManager.Delete([]client.Object{ns})).To(Succeed())
+	})
+})
+
+func writeTrafficResults(resultsFile *os.File, homeDir, filename, testname string) {
+	file := fmt.Sprintf("%s/%s", homeDir, filename)
+	content, err := os.ReadFile(file)
+	Expect(err).ToNot(HaveOccurred())
+
+	formattedContent := fmt.Sprintf("%s:\n\n```text\n%s```\n", testname, string(content))
+	Expect(framework.WriteContent(resultsFile, formattedContent)).To(Succeed())
+}
+
+func writeErrorLogs(resultsFile *os.File, pod, container string) {
+	logReq := clientGoClient.CoreV1().Pods(ngfNamespace).GetLogs(pod, &core.PodLogOptions{Container: container})
```

Comment:
does it give you all logs for 4 days or only recent logs?

Reply:
It should get all logs, same as `kubectl logs`.

Reply:
based on this https://kubernetes.io/docs/concepts/cluster-administration/logging/#log-rotation , only the latest logs are given

Reply:
Oh I see, I must've read something different.

Reply:
Well, the other option is enable google logging and this test now requires more user steps and less automation...

Reply:
> Well, the other option is enable google logging and this test now requires more user steps and less automation...

is it possible to use gcloud API to run those log queries?

Reply:
I've updated to use gcloud to gather the logs. However, it will gather all logs that match the criteria even for old clusters with the same name. Trying to figure out the best way to filter out everything except the current cluster. Thinking about timestamps, though that may get ugly. Maybe labels in some way.

Reply:
Pod names should be different (unique)? Could you filter by pod names?

Reply:
I'm actually going to deploy this with a different release name, which will allow me to filter by the label that contains that name.

The problem with pod names is that the pods could go up and down throughout the 4 days, so we won't know what the old names were.